olive/evaluator/olive_evaluator.py

```diff
@@ -468,19 +430,53 @@ def _inference(
             execution_providers=execution_providers,
         )
 
-        OnnxEvaluator.disable_ort_fallback(session, execution_providers)
-
         io_config = model.get_io_config()
 
         preds = []
         targets = []
         logits = []
         logits_dict = collections.defaultdict(list)
         output_names = io_config["output_names"]
+        io_bind = self.io_bind_enabled(metric, model.inference_settings)
+        device = "cuda" if device == "gpu" else "cpu"
+        if io_bind and device == "cuda":
+            io_binding = session.io_binding()
+            input_data, _ = next(iter(dataloader))
```

Comment:
We may need to rewind the dataloader after `next(iter(dataloader))`.

Otherwise, the next `for input_data, labels in dataloader` will start from second one of the full data.

Reply:
I don't think that's needed? We expect the dataloader to be similar to torch dataloader. There is no concept of rewind like there is for the onnxruntime quantization calibration dataloader. 

For this dataloader, the expectation is that every new `iter(dataloader)` call on the dataloader starts the iteration from the start. 
The for loop is also an instance of a new `iter` over the dataloader.

Reply:
I am wrong.  No need to remind.